libs/dmlf/include/dmlf/stochastic_reception_algorithm.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <memory>
+
+#include "dmlf/shuffle_algorithm_interface.hpp"
+
+namespace fetch {
+namespace dmlf {
+
+class StochasticReceptionAlgorithm : public ShuffleAlgorithmInterface
```

Comment:
this class is a little confusing to me. It looks like it's only functionality is to return broadcast_proportion_?
Also doesn't the shuffle_algorithm_interface now live in deprecated?

Reply:
Yeah, I've undeprecated the shuffle